content/contexts/experimenterDetails.tsx

```diff
@@ -0,0 +1,77 @@
+import PropTypes from "prop-types";
+import React from "react";
+
+interface ExperimenterData {
+  publicDescription: string;
+  proposedStartDate: Date;
+  proposedEndDate: Date;
+  proposedDuration: number;
+  startDate?: Date;
+  endDate?: Date;
+  variants: string[];
+}
+
+interface ExperimenterState {
+  data: ExperimenterData;
+}
+
+export const INITIAL_EXPERIMENTER_DATA = {
+  publicDescription: "",
+  proposedStartDate: new Date(),
+  proposedEndDate: new Date(),
+  proposedDuration: 0,
+  startDate: null,
+  endDate: null,
+  variants: [],
+};
+
+const initialState: ExperimenterState = {
+  data: INITIAL_EXPERIMENTER_DATA,
+};
+
+export const experimenterDetailsContext = React.createContext(initialState);
+const { Provider } = experimenterDetailsContext;
+
+export const ACTION_UPDATE_DATA = "UPDATE_DATA";
+
+function reducer(state, action): ExperimenterState {
+  switch (action.type) {
+    case ACTION_UPDATE_DATA:
+      return {
+        ...state,
+        data: {
+          ...state.data,
+          ...action.data,
+        },
+      };
+
+    default:
+      return state;
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export function ExperimenterDetailsProvider({ children, data }) {
+  /** @type {[React.ReducerState<any>, React.Dispatch<React.ReducerAction<any>>]} */
+  const [state, dispatch] = React.useReducer(reducer, initialState);
+
+  React.useEffect(() => {
+    dispatch({
+      type: ACTION_UPDATE_DATA,
+      data,
+    });
+  }, [data]);
+
+  // @ts-ignore
+  return <Provider value={state}>{children}</Provider>;
+}
+
+ExperimenterDetailsProvider.propTypes = {
```

Comment:
We probably shouldn't be mixing proptypes and Typescript.

Reply:
I think I'll need some guidance, I was looking at `normandy-devtools/content/contexts/recipeDetails.js ` and https://github.com/mozilla-extensions/normandy-devtools/blob/c8d75d4650216e451a23fe176c1fff27735bcdb5/content/contexts/environment.tsx#L256-L258

Reply:
Oops, I guess I missed that in my conversion. The pattern I've using has been to leverage the `React.FunctionalComponent` (or `React.FC` for short) which can be parameterized on the type of its props. So in this case it would be something like

```ts
const ExperimenterDetailsProvider: React.FC<ExperimenterState> = ({ children, data }) => {
	// ...
}
```

`children` is automatically added to the type of the props by `React.FC`. This results in bad props being a build time problem instead of a quiet runtime problem.